Experimental/DistributedAuthoritySample/Assets/Scripts/Gameplay/EnvironmentTransform.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using Unity.Netcode;
+using Unity.Netcode.Components;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.SocialHub.Gameplay
+{
+    class EnvironmentTransform : NetworkTransform, IOwnershipRequestable, IGameplayEventInvokable
+    {
+        public event Action<NetworkObject, NetworkObject.OwnershipRequestResponseStatus> OnNetworkObjectOwnershipRequestResponse;
+
+        public event Action<NetworkObject, GameplayEvent> OnGameplayEvent;
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+
+            NetworkObject.OnOwnershipRequested += OnOwnershipRequested;
+            NetworkObject.OnOwnershipRequestResponse += OnOwnershipRequestResponse;
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            if (NetworkObject)
+            {
+                NetworkObject.OnOwnershipRequested -= OnOwnershipRequested;
+                NetworkObject.OnOwnershipRequestResponse -= OnOwnershipRequestResponse;
+            }
+
+            OnGameplayEvent?.Invoke(NetworkObject, GameplayEvent.Despawned);
+        }
+
+        protected override void OnOwnershipChanged(ulong previous, ulong current)
+        {
+            base.OnOwnershipChanged(previous, current);
+
+            OnGameplayEvent?.Invoke(NetworkObject, GameplayEvent.OwnershipChange);
+        }
+
+        // note: invoked on owning client
+        bool OnOwnershipRequested(ulong clientRequesting)
```

Comment:
The only very minor nit-pick type of thing would be adding `private` in front of some of the methods and properties throughout the various classes.